### PR TITLE
Add Sitemap.xml functionality.

### LINKED ISSUE #544: Sitemap.xml. The documentation site should have it.
For better indexing by the googles and such.

### PR DESCRIPTION
**Pull request by bobdenotter**
Fixes #544

### CODE REVIEW

File: src/Controllers.php [hunk 1]
```diff
@@ -106,6 +113,40 @@ public function cheatsheet(Version $version)
...
+     * Controller for sitemap (single)
+     *
+     * @param Version $version
+     *
+     * @return mixed
+     */
+    public function sitemap(Version $version)
+    {
+        $twigVars = [
+            'menu' => $version->getMenu(),
```
Comment (CarsonF) [inline src/Controllers.php]: Could you use `$version->getPage('')` here instead. I'm actually trying to kill `getMenu`.

Maybe just change the key to `page` or `root` or `rootPage`.

File: view/templates/_macros.twig [hunk 2]
```diff
@@ -56,3 +56,11 @@
         </ul>
     </li>
 {% endmacro %}
+
+{% macro sitemap(page) %}
+    {% from _self import sitemap %}
+    <url><loc>https://docs.bolt.cm{{ page.url }}</loc></url>
+    {% for sub in page.children if sub is not empty %}
```
Comment (CarsonF) [inline view/templates/_macros.twig]: The `if` isn't needed here. And would change to `page.subPages` with menu change.

File: src/Controllers.php [hunk 3]
```diff
@@ -106,6 +113,40 @@ public function cheatsheet(Version $version)
...
+
+    /**
+     * Controller for sitemap-sitemap (multiple)
+     *
+     * @return mixed
+     */
+    public function sitemap_list()
+    {
+        $twigVars = [
+            'versions' => array_keys($this->app['documentation']->getVersions()),
```
Comment (CarsonF) [inline src/Controllers.php]: `array_keys` isn't needed here. `Version` has a `__toString` method.

File: src/Controllers.php [hunk 4]
```diff
@@ -33,6 +33,13 @@ public function connect(Silex\Application $app)
             ->value('page', '')
             ->assert('page', '.*');
 
+        $ctr->get('/sitemap/{version}.xml', [$this, 'sitemap'])
+            ->bind('sitemap')
+            ->value('version', '');
```
Comment (CarsonF) [inline src/Controllers.php]: `/sitemap/.xml`? I don't think you can or need to set version's default value to empty string here.

File: view/templates/_macros.twig [hunk 5]
```diff
@@ -56,3 +56,11 @@
         </ul>
     </li>
 {% endmacro %}
+
+{% macro sitemap(page) %}
+    {% from _self import sitemap %}
+    <url><loc>https://docs.bolt.cm{{ page.url }}</loc></url>
```
Comment (CarsonF) [inline view/templates/_macros.twig]: Do `loc`'s need to be absolute?

File: src/Controllers.php (hunk 4 above)
Comment (bobdenotter) [inline src/Controllers.php]: Can I remove this one too, then?

https://github.com/bolt/docs/blob/site/src/Controllers.php#L32

Comment (CarsonF) [inline src/Controllers.php]: No because that indirectly defines the homepage.

**Comment by bobdenotter** [comment]
@CarsonF All your suggestions should be addressed now. :-)

**Comment by bobdenotter** [comment]
Well, It _is_ the tree with links to all of the pages, so I figured the name was apt.. 

I'll change it though, 10 second fix. :-)

**Pull request merged**